test/topology_custom/test_mv_large_inserts.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+
+logger = logging.getLogger(__name__)
+
+async def insert_with_concurrency(cql, concurrency):
+    def serialize_int(i):
+        return Int32Type.serialize(i, cql.cluster.protocol_version)
+    def get_replicas(key):
+        return cql.cluster.metadata.get_replicas("ks", serialize_int(key))
+    for key in range(100):
+        logger.info(f"Starting writes with concurrency {concurrency}")
+        async def do_inserts(i: int):
+            key = i
+            local_node = get_replicas(key)[0]
+            while local_node == get_replicas(i)[0]:
+                i = i + concurrency
+            await cql.run_async(f"INSERT INTO ks.tab (key, c, v) VALUES ({key}, {i}, '{10000 * "a"}')")
+
+        tasks = [asyncio.create_task(do_inserts(i)) for i in range(concurrency)]
+        await asyncio.gather(*tasks)
+        logger.info(f"Finished writes with concurrency {concurrency}")
+
+async def wait_for_views(cql, mvs_count, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        logger.info(f"Views built: {done[0][0]}")
+        if done[0][0] == node_count * mvs_count:
```

Comment:
Unless I'm missing something, this test looks very fragile - what if other views were built in the past on other tables? I think you should look for the views of one particular table - not just for all the views in the system.

Yes, I know a lot of topology tests assume it uses a fresh cluster (so in particular this code will work). But I think this (assuming a fresh cluster) is a mistake we'll regret (I already regret it), and moreover, in the future we might (?) have internal materialized views or something.

Reply:
While I'm changing this test anyway, I don't think we'll be able to prepare a (minimal) unit test without a fresh cluster - if we want the test to be quick and repeatable, we need to start with some given size of the backlog (empty) so that we know how much data should we use.
Even if it was possible to make a consistent test that can run in an existing cluster, I'm not sure if additionally preparing a framework for "testing the test" is worth it for a single test.